ui/src/app/integrations/list-page/list-page.component.ts

```diff
@@ -72,6 +71,16 @@ export class IntegrationsListPage implements OnInit {
           null,
           []
         );
+      } else if (status === 400) {
+        const r = JSON.parse(response);
```

Comment:
Naming things just as `r` is considered a code smell in JavaScript-land nowadays. It hinders code comprehension in long scripts. Perhaps `jsonResponse` would work better here ;-)

Reply:
I tend to agree when you have a large code block, but in small code blocks like this, shorter variable names seem easier to track.

Reply:
but fixed